soap/src/java/com/zimbra/soap/admin/message/HabOrgUnitRequest.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ * Zimbra Collaboration Suite Server
+ * Copyright (C) 2018 Synacor, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software Foundation,
+ * version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ * ***** END LICENSE BLOCK *****
+ */
+package com.zimbra.soap.admin.message;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlEnum;
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.XmlType;
+
+import com.google.common.base.MoreObjects;
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.soap.AdminConstants;
+import com.zimbra.soap.admin.type.DomainSelector;
+
+/**
+ * @author zimbra
+ *
+ */
+/**
+ * @zm-api-command-auth-required true
+ * @zm-api-command-admin-auth-required true
+ * @zm-api-command-description Create/Rename/Delete Hab Organizational Unit
+ */
+@XmlAccessorType(XmlAccessType.NONE)
+@XmlRootElement(name = AdminConstants.E_HAB_ORG_UNIT_REQUEST)
+@XmlType(propOrder = {})
+public class HabOrgUnitRequest {
+
+    @XmlEnum
+    public enum HabOp {
+        // case must match
+        create, rename, delete;
+
+        public static HabOp fromString(String s) throws ServiceException {
+            try {
+                return HabOp.valueOf(s);
+            } catch (IllegalArgumentException e) {
+                throw ServiceException.INVALID_REQUEST("unknown key: "+s, e);
+            }
+        }
+    }
+
+    
+    /**
+     * @zm-api-field-description Specifies the domain under which Hab org unit
+     *                           is created
+     */
+    @XmlElement(name = AdminConstants.E_DOMAIN, required = true)
+    private DomainSelector domainSelector;
+
+    /**
+     * @zm-api-field-description HAB org unit name
+     */
+    @XmlAttribute(name = AdminConstants.A_NAME /* op */, required = true)
```

Comment:
please correct the comment - op. same for other members below.

Reply:
Fixed